### ISSUE TITLE
In some applications it doesn't read the column unless tab and shift+tab are done

### ISSUE THREAD

**Issue by Lo-lo78** [ABuffEr/columnsReview#33]
@ABuffEr 
Hi Alberto, I noticed that in some applications the column is not read as if the addon was not on focus and I have to do something to unblock the problem.
I usually tab and shift+tab to get back into the list and the addon starts working again.
It always works in Explorer!

If you want to reproduce the problem:
1.
I open Reaper F4 and go to action list and try to read columns. It does not work.
2.
With Emule if I read a column for the first time it makes the sound of Windows as if the shortcut weren't there. I do tab and shift+tab and now column reading works.

I haven't tried other programs but it seems that the problem repeats itself in various environments.
Greetings!

**Comment by ABuffEr** [comment]
Hi,
what NVDA version are you using?
Because some recent alphas have strange problems with gestures (not directly related to ColumnsReview).

**Comment by Lo-lo78** [comment]
Sorry, I didn't tell you the Columns Review version either.
I am using NVDA: 2023.1 e Columns Review 20230602-dev.

**Comment by ABuffEr** [comment]
Hi,
I don't use Reaper. Can you detail steps in eMule? I noticed nothing there for now...
Please report also your add-on configuration.

**Comment by Lo-lo78** [comment]
@ABuffEr 
I can give you a more immediate example.
With the NVDA version: alpha-28584,5c4e6e1f if I open the Add-on Store window and search with the letter C Columns Review and try to read the columns it doesn't work.
If I do tab or shift+tab it starts working.
With Emule it does it in some cases and not always. With the Store, the bug always shows up in both NVDA stable and alpha releases.
I hope I was helpful.

Greetings!

**Comment by Lo-lo78** [comment]
@ABuffEr 
I try to create a portable version of NVDA with only Columns Review so I check if it is a conflict with another addon.
I'll let you know if the problem remains.

**Comment by ABuffEr** [comment]
Ah, I see. Probably due to dynamic list populating. I'm already working on this, but thanks for the good testing example!

**Comment by Lo-lo78** [comment]
@ABuffEr 
Perhaps you have already identified the problem.
I did other tests.
I have installed the portable alpha version of NVDA and tried Columns Review 20230602-dev and in the Store it gives this problem. It doesn't read columns unless you unlock the addon.
I already reported it to you but with explorer it works perfectly.
So it doesn't happen in all situations.
Thank you!

**Comment by ABuffEr** [comment]
Hi @Lo-lo78,
try the [new build.](https://github.com/ABuffEr/columnsReview/releases/download/20230717-dev/columnsReview-20230717-dev.nvda-addon)

**Comment by Lo-lo78** [comment]
@ABuffEr 
I just tried the development version and for the NVDA store there are no problems. Works well.
In all other situations where I tried it gives the same problem.
You have to unlock the addon for it to work.
The apps I've tried are:
Emule and Reaper.
In these two applications the addon has remained the same.
I don't know what it's about!

**Comment by ABuffEr** [comment]
Hi @Lo-lo78,
can you try the [new build?](https://github.com/ABuffEr/columnsReview/releases/download/v5.3.20230812/columnsReview-5.3.20230812.nvda-addon)
Yes, I changed version naming for dev builds, to publish them on add-on store.

**Comment by amirsol81** [comment]
@ABuffEr With the latest development build, I have the same issue in RSSOwlnix on Windows 10. Though the app is maximized, I need to bring up another window, say the list of TTS engines by pressing Control+NVDA+S, first and dismiss it in order for ColumnsReview to function. Otherwise, pressing Control+NVDA+numbers in the list of articles forces NVDA to remain silent.

**Comment by Lo-lo78** [comment]
@ABuffEr 
Hi Alberto, unfortunately on some applications it gives the same problem.
Nothing has changed.
For the moment I have only tried Emule. If you want, I'll point out all the applications that I encounter with the same problem.
Thanks for what you are doing.

**Comment by amirsol81** [comment]
@ABuffEr Update: this is also affecting RSSOwlnix on Windows 11, so it isn't limited to Win 10. So if I open a folder, Control+NVDA+numbers fail to announce column headers unless I bring up an extra dialog, close it, and return to the list of articles. And this happens with all folders as they gain focus, so isn't limited to the first folder in the session. I've also made sure RSSOwlnix is maximized. Quite strange...

**Comment by ABuffEr** [comment]
Hi @amirsol81,
that is another issue, even if it's trange. You verified that nothing change simply restarting NVDA?
Tomorrow I'll run Win11 in a VM and investigate about desktop.

**Comment by amirsol81** [comment]
@ABuffEr Thanks. I tried the Maximum power setting - it was selected anyway on both Win 10 and 11 machines running the latest NVDA alpha which is almost the equivalent of NVDA 2023.2 Beta 2, and I still get the same. ColumnsReview doesn't respond to Control+NVDA+numbers in the list of articles in RSSOwlnix unless a dialog is brought up and dismissed first. And as I enter/open each folder/RSS feed, the same happens.

**Comment by amirsol81** [comment]
@ABuffEr I'd also like to know your take on this one.

**Comment by ABuffEr** [comment]
Hopefully, I found the lacking brick: sometimes NVDA receives a gainFocus event on list, instead of focusEntered, even when list is already populated. In RSSOwlnix seems ok now. I'll publish the new build tomorrow.

**Comment by amirsol81** [comment]
@ABuffEr That's fantastic! Looking forward to it, and thanks for your continuous maintenance of such a useful add-on.

**Comment by ABuffEr** [comment]
Hi,
try the new build, [from here](https://github.com/ABuffEr/columnsReview/releases/download/v5.3.20230821/columnsReview-5.3.20230821.nvda-addon) (later on add-on store if it finally works).

**Comment by amirsol81** [comment]
@ABuffEr Thanks! This fixes the RSSOwlnix issue.

**Comment by Lo-lo78** [comment]
@ABuffEr 
Perfect!
Works well...
👍👍👍

**Comment by ABuffEr** [comment]
Ok, thanks to all! I finally close the issue as completed.
